pkg/engine/virtualmachines.go

```diff
@@ -563,3 +565,9 @@ func getVaultSecretGroup(linuxProfile *api.LinuxProfile) []compute.VaultSecretGr
 	}
 	return vaultSecretGroups
 }
+
+func addCustomTagsToVM(cs *api.ContainerService, vm *compute.VirtualMachine) {
+	for key, value := range cs.Properties.MasterProfile.CustomVMTags {
```

Comment:
where does the equivalent happen for agentpool VMs?

Reply:
here you could pass in the tags instead of passing in the full container service

`func addCustomTagsToVM(tags map[string]string, vm *compute.VirtualMachine) {`

that way it's reusable for agents

Reply:
Sure, I'll fix that!

Reply:
Done!